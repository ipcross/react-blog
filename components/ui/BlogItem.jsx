import React from 'react';
import Image from './Image';
import TextBox from './TextBox';

const BlogItem = (props) => (
  <div>
      <Image image={props.post.image} />
      <TextBox>{props.post.text}</TextBox>
  </div>
)

export default BlogItem;
