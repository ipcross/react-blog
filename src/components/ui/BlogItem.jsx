import React from 'react';
import PropTypes from 'prop-types';
import Link from 'components/elements/Link';
import { postsPath } from 'helpers/routes';

import Image from 'components/ui/Image.jsx';
import PostDetails from 'components/ui/PostDetails.jsx';
import LikeContainer from 'containers/LikeContainer';

const BlogItem = (props) => (
  <div>
    <Image {...props.post.image} />
    <Link to={postsPath(props.post.id)}>{props.post.text}</Link>
    <PostDetails {...props.post.details} />
    <LikeContainer postId={props.post.id} />
  </div>
);

BlogItem.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.number,
    text: PropTypes.string,
    image: Image.propTypes.image,
    details: PostDetails.propTypes.details,
    likes: PropTypes.number
  })
};

export default BlogItem;
