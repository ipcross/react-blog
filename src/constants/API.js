export const APP_ROOT = 'http://192.168.23.148:3000';
export const API_ROOT = 'http://192.168.23.148:3001';
