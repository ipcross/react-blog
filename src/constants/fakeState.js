export const fakeState = {
  posts: {
    isFetching: false,
    error: false,
    entries: {
      posts: [
        {
          id: 10,
          title: 'Violets',
          description: 'Narcissus were well known in ancient civilisation, both medicinally and botanically, but formally described by Linnaeus in his Species Plantarum (1753). The genus is generally considered to have about ten sections with approximately 50 species.',
          meta: {
            likes: 9,
            author: 'user_10',
            createdAt: '2017-07-27T14:54:32.006Z'
          },
          image: {
            src: 'http://192.168.23.148:3001/flower10.jpg'
          },
          comments: [],
          files: []
        }
      ],
      meta: {
        current_page: 1,
        next_page: 2,
        prev_page: null,
        total_pages: 2,
        total_count: 10,
        per_page: 5
      }
    }
  },
  post: {
    isFetching: false,
    error: false,
    entry: {
      post: {
        id: 10,
        title: 'Violets',
        description: 'Narcissus were well known in ancient civilisation, both medicinally and botanically, but formally described by Linnaeus in his Species Plantarum (1753). The genus is generally considered to have about ten sections with approximately 50 species.',
        meta: {
          likes: 9,
          author: 'user_10',
          createdAt: '2017-07-27T14:54:32.006Z'
        },
        image: {
          src: 'http://192.168.23.148:3001/flower10.jpg'
        },
        comments: [],
        files: []
      }
    }
  },
  form: {}
};
