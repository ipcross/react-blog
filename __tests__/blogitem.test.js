import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';

import Link from 'components/elements/Link';
import { fakeStore } from 'helpers/fakeStore'
import { fakeState } from 'constants/fakeState'

import BlogItem from 'components/ui/BlogItem';

describe('BlogItem', () => {
  let state = fakeState;
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={fakeStore(state)}><BlogItem /></Provider>, div);
  });
  
  describe('render', () => {
    it('should render the title', () => {
      const itemProps = { title: 'Hello, World!', id: 5 };

      const item = shallow(<BlogItem post={itemProps} />);
      const header = <Link to="/posts/5">Hello, World!</Link>;

      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const itemProps = {
        title: 'Hello, World!',
        id: 5,
        image: {
          alt: 'Hello, World',
          src: '/images/World.jpg'
        }
      };

      const item = shallow(<BlogItem post={itemProps} />);

      expect(item).toMatchSnapshot();
    })
  });
});