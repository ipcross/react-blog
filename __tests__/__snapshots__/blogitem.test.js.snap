// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem render should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Item>
    <Image
        alt="Hello, World"
        src="/images/World.jpg"
        style={
            Object {
                "float": "left",
                "height": 100,
                "marginRight": 10,
                "width": 150,
              }
        }
    />
    <ItemContent>
        <ItemHeader
            as="h2"
        >
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/posts/5"
            >
                Hello, World!
            </Link>
        </ItemHeader>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/posts/5/edit"
        >
            Edit
        </Link>
        <ItemDescription>
            <TextBox>
                Default text for TextBox
            </TextBox>
        </ItemDescription>
        <br />
        <ItemMeta>
            <PostDetails
                author="Default Author"
            />
            <Connect(Likes)
                postId={5}
            />
        </ItemMeta>
    </ItemContent>
</Item>,
  "nodes": Array [
    <Item>
      <Image
            alt="Hello, World"
            src="/images/World.jpg"
            style={
                  Object {
                        "float": "left",
                        "height": 100,
                        "marginRight": 10,
                        "width": 150,
                      }
            }
      />
      <ItemContent>
            <ItemHeader
                  as="h2"
            >
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/posts/5"
                  >
                        Hello, World!
                  </Link>
            </ItemHeader>
            <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/posts/5/edit"
            >
                  Edit
            </Link>
            <ItemDescription>
                  <TextBox>
                        Default text for TextBox
                  </TextBox>
            </ItemDescription>
            <br />
            <ItemMeta>
                  <PostDetails
                        author="Default Author"
                  />
                  <Connect(Likes)
                        postId={5}
                  />
            </ItemMeta>
      </ItemContent>
</Item>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        post={
                Object {
                        "id": 5,
                        "image": Object {
                          "alt": "Hello, World",
                          "src": "/images/World.jpg",
                        },
                        "title": "Hello, World!",
                      }
        }
/>,
      "_debugID": 55,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "post": Object {
            "id": 5,
            "image": Object {
              "alt": "Hello, World",
              "src": "/images/World.jpg",
            },
            "title": "Hello, World!",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Item>
          <Image
                    alt="Hello, World"
                    src="/images/World.jpg"
                    style={
                              Object {
                                        "float": "left",
                                        "height": 100,
                                        "marginRight": 10,
                                        "width": 150,
                                      }
                    }
          />
          <ItemContent>
                    <ItemHeader
                              as="h2"
                    >
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        Hello, World!
                              </Link>
                    </ItemHeader>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/posts/5/edit"
                    >
                              Edit
                    </Link>
                    <ItemDescription>
                              <TextBox>
                                        Default text for TextBox
                              </TextBox>
                    </ItemDescription>
                    <br />
                    <ItemMeta>
                              <PostDetails
                                        author="Default Author"
                              />
                              <Connect(Likes)
                                        postId={5}
                              />
                    </ItemMeta>
          </ItemContent>
</Item>,
        "_debugID": 56,
        "_renderedOutput": <Item>
          <Image
                    alt="Hello, World"
                    src="/images/World.jpg"
                    style={
                              Object {
                                        "float": "left",
                                        "height": 100,
                                        "marginRight": 10,
                                        "width": 150,
                                      }
                    }
          />
          <ItemContent>
                    <ItemHeader
                              as="h2"
                    >
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        Hello, World!
                              </Link>
                    </ItemHeader>
                    <Link
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="/posts/5/edit"
                    >
                              Edit
                    </Link>
                    <ItemDescription>
                              <TextBox>
                                        Default text for TextBox
                              </TextBox>
                    </ItemDescription>
                    <br />
                    <ItemMeta>
                              <PostDetails
                                        author="Default Author"
                              />
                              <Connect(Likes)
                                        postId={5}
                              />
                    </ItemMeta>
          </ItemContent>
</Item>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    post={
        Object {
            "id": 5,
            "image": Object {
              "alt": "Hello, World",
              "src": "/images/World.jpg",
            },
            "title": "Hello, World!",
          }
    }
/>,
}
`;
